torch/nn/parallel/distributed.py

```diff
@@ -800,33 +835,32 @@ def forward(self, *inputs, **kwargs):
             else:
                 output = self.module(*inputs, **kwargs)
 
+            disable = False
             if torch.is_grad_enabled() and self.require_backward_grad_sync:
                 self.require_forward_param_sync = True
-                # We'll return the output object verbatim since it is a freeform
-                # object. We need to find any tensors in this object, though,
-                # because we need to figure out which parameters were used during
-                # this forward pass, to ensure we short circuit reduction for any
-                # unused parameters. Only if `find_unused_parameters` is set.
-                if self.find_unused_parameters and not self.static_graph:
-                    # Do not need to populate this for static graph.
-                    self.reducer.prepare_for_backward(list(_find_tensors(output)))
-                else:
-                    self.reducer.prepare_for_backward([])
+                if disable or not self.find_unused_parameters or self.static_graph:
+                    if self.find_unused_parameters:
+                        self.reducer.prepare_for_backward(list(_find_tensors(output)))
```

Comment:
in which case this will happen?

Reply:
Yeah, with this change we should never need to call `self.reducer.prepare_for_backward(list(_find_tensors(output)))`. The suggestion below is much clearer.